src/auth/router.js

```diff
@@ -0,0 +1,154 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const jsonwebtoken = require('jsonwebtoken');
+const crypto = require('crypto');
+
+const db = require('../database/connection');
+const sendEmail = require('../utils/emailhelper');
+
+const router = express.Router();
+
+const executeQuery = ({ query, params }) => {
+  return new Promise((resolve, reject) => {
+    db.query(query, params, (err, rows) => {
+      if (err) {
+        reject(err);
+      }
+
+      if (rows.length === 0) {
+        resolve(null);
```

Comment:
Please consider for query the case get list (return empty array) and get one (returl null)